lib/pbench/server/database/models/datasets.py

```diff
@@ -434,34 +341,12 @@ def stem(path: Union[str, Path]) -> str:
         Returns:
             The stripped "stem" of the dataset
         """
-        p = Path(path)
+        p = Path(str(path))
```

Comment:
I actually meant to try backing this out, but forgot. I figured it'd trigger you, but hoped nobody would notice. Basically, I ran into unit test problems galore for a while, which spontaneously went away ... I mean, really crazy things like new proxy objects appearing in queries before they were even added. One of the minor symptoms was a complaint here, possibly because there was a `MockPath` (even though there was an `__str__` on it), and adding this fixed the problem. I hoped that would resolve itself when the craziness passed, but never got around to testing without it. It's harmless ... but if I have to make another pass anyway I might ratchet up my courage and try it.

Reply:
> this started failing because `path` is not `str`, `byte`, or `Path`.

Um...I wonder exactly what it **_is_**, then!  😨 

> hacking the product code for a unit test bugs me

Me, too....

> I'm not inclined to wrestle this lion any further in this context...

Okie.  😞

Reply:
> > this started failing because `path` is not `str`, `byte`, or `Path`.
> 
> Um...I wonder exactly what it **_is_**, then! fearful

It's a `MockPath`; but if there's a way to make the real `Path` recognize this as a "path-like object" I was unable to discover how. Simply having `__str__` isn't enough. The weird thing is that neither the test case nor the code here is new, and I couldn't see what changed.

Reply:
It needs to be a [path-like object](https://docs.python.org/3/glossary.html#term-path-like-object), which means it needs to be derived from [os.PathLike](https://docs.python.org/3/library/os.html#os.PathLike) and/or implement the [`__fspath__()`](https://docs.python.org/3/library/os.html#os.PathLike.__fspath__) method.

Reply:
Now that along is worth another commit ...